unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/GotoPanelHUD/GotoPanelHUDView.cs

```diff
@@ -0,0 +1,96 @@
+    [SerializeField] public GameObject container;
+    [SerializeField] internal TextMeshProUGUI sceneTitleText;
+    [SerializeField] internal TextMeshProUGUI sceneOwnerText;
+    [SerializeField] internal Sprite scenePreviewFailImage;
+    [SerializeField] internal RawImageFillParent scenePreviewImage;
+    [SerializeField] internal GameObject loadingSpinner;
+    [SerializeField] internal Button closeButton;
+    [SerializeField] internal Button cancelButton;
+    [SerializeField] internal ShowHideAnimator contentAnimator;
+
+    private ParcelCoordinates targetCoordinates;
+
+    AssetPromise_Texture texturePromise = null;
+
+    private void Start()
+    {
+        teleportButton.onClick.RemoveAllListeners();
+        teleportButton.onClick.AddListener(TeleportTo);
+        closeButton.onClick.AddListener(OnClosePressed);
+        cancelButton.onClick.AddListener(OnClosePressed);
+        container.SetActive(false);
+        contentAnimator.OnWillFinishHide += (animator) => Hide();
+    }
+
+    private void TeleportTo()
+    {
+        WebInterface.GoTo(targetCoordinates.x, targetCoordinates.y); 
+        OnClosePressed();
+    }
+
+    public void SetVisible(bool isVisible) 
+    {
+        container.SetActive(isVisible);
+        contentAnimator.Show(!isVisible);
+        loadingSpinner.SetActive(isVisible);
+        scenePreviewImage.texture = null;
+    }
+
+    public void SetPanelInfo(ParcelCoordinates parcelCoordinates)
+    {
+        MinimapMetadata.MinimapSceneInfo sceneInfo = MinimapMetadata.GetMetadata().GetSceneInfo(parcelCoordinates.x, parcelCoordinates.y);
+        if (sceneInfo != null)
+        {
+            sceneTitleText.text = sceneInfo.name;
+            sceneOwnerText.text = sceneInfo.owner;
+            SetParcelImage(sceneInfo);
+        }
+        else
+        {
+            sceneTitleText.text = "Untitled Scene";
+            sceneOwnerText.text = "Unknown";
+            DisplayThumbnail(scenePreviewFailImage.texture);
+        }
+        targetCoordinates = parcelCoordinates;
+        panelText.text = $"{parcelCoordinates.x},{parcelCoordinates.y}";
+    }
+
+    private void SetParcelImage(MinimapMetadata.MinimapSceneInfo sceneInfo)
+    {
+        DisplayThumbnail(scenePreviewFailImage.texture);
+        if (!string.IsNullOrEmpty(sceneInfo.previewImageUrl))
+        {
+            texturePromise = new AssetPromise_Texture(sceneInfo.previewImageUrl, storeTexAsNonReadable: false);
+            texturePromise.OnSuccessEvent += (textureAsset) => { DisplayThumbnail(textureAsset.texture); };
+            texturePromise.OnFailEvent += (textureAsset, error) => { DisplayThumbnail(scenePreviewFailImage.texture); };
+            AssetPromiseKeeper_Texture.i.Keep(texturePromise);
+        }
+    }
+
+    private void DisplayThumbnail(Texture2D texture)
+    {
+        loadingSpinner.SetActive(false);
+        scenePreviewImage.texture = texture;
+    }
+
+    private void OnClosePressed()
```

Comment:
Same as with teleport, the controller should do most of this logic by setting `gotoPanelVisible` and calling a `view.Hide` or something similar.